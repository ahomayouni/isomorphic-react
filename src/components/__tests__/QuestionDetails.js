import { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";
import renderer from "react-test-renderer";
import React from "react";

// Map state to props sample
describe(`The Question Detail Component`, () => {
  describe(`The Container Element`, () => {
    describe("mapStateToPros", () => {
      it("should map the state to props correctly", () => {
        const sampleQuestion = {
          question_id: 42,
          body: "Something",
        };
        const appState = {
          questions: [sampleQuestion],
        };
        const ownProps = {
          question_id: 42,
        };
        const componentState = mapStateToProps(appState, ownProps);
        expect(componentState).toEqual(sampleQuestion);
      });
    });
  });

  describe("the display element", () => {
    it("should not regress", () => {
      const tree = renderer.create(
        <QuestionDetailDisplay
          title="title"
          body="body"
          answer_count={0}
          tags={["hitchhiking"]}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
