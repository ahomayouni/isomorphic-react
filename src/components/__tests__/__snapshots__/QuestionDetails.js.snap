// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component the display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    title
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hitchhiking
        </code>
      </div>
    </div>
    <p>
      body
    </p>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
